### PR TITLE
[Feature][AMS]: support show table info for native iceberg in ams dashboard

### LINKED ISSUE #531: [Feature][AMS]: support show table info for native iceberg in ams dashboard
### Description

support show table schema and fileinfo, snpashotinfo of native iceberg table. 
Attention: when get native iceberg table fileinfo by iceberg interface we should support get partial file info from iceberg manifest file to avoid oom and and show pages

### Use case/motivation

_No response_

### Describe the solution

_No response_

### Subtasks

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hameizi**
## Why are the changes needed?
resolve #531 

## Brief change log

add logic: get native iceberg table snapshot info

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: core/src/main/java/com/netease/arctic/utils/SnapshotFileUtil.java [hunk 1]
```diff
@@ -126,4 +130,55 @@ private static void getDeleteFiles(
       LOG.error("close manifest file error", exception);
     }
   }
+
+  public static void getNativeIcebergDeleteFiles(
```
Comment (zhoujinsong) [inline core/src/main/java/com/netease/arctic/utils/SnapshotFileUtil.java:134]: How about rename method name to `getIcebergDeleteFiles` and make `getDeleteFiles` depende on this one?

**Review by zhoujinsong — COMMENTED** [review]
@hameizi  I left some comments. PTAL.

**Comment by zhoujinsong** [comment]
@hameizi Should the WEB UI hidden `Files` tabls for iceberg tables?

**Comment by hameizi** [comment]
> @hameizi Should the WEB UI hidden `Files` tabls for iceberg tables?

This PR not support native iceberg file info in web ui

File: core/src/main/java/com/netease/arctic/utils/SnapshotFileUtil.java (hunk 1 above)
Comment (hameizi) [reply core/src/main/java/com/netease/arctic/utils/SnapshotFileUtil.java:134]: There is some difference between arctic and iceberg delete file. Arctic eq-delete file is datafile and all delete file is pos-delete file, but in iceberg there are both eq-delete and pos-delete type exist in table.

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/model/AMSDataFileInfo.java [hunk 2]
```diff
@@ -18,10 +18,9 @@
 
 package com.netease.arctic.ams.server.model;
 
-import com.netease.arctic.ams.api.DataFileInfo;
+import com.netease.arctic.ams.api.DataFile;
```
[the remark this answers is not in the document]
Comment (hameizi) [reply ams/ams-server/src/main/java/com/netease/arctic/ams/server/model/AMSDataFileInfo.java]: remove in new commit.

File: core/src/main/java/com/netease/arctic/utils/SnapshotFileUtil.java (hunk 1 above)
Comment (zhoujinsong) [reply core/src/main/java/com/netease/arctic/utils/SnapshotFileUtil.java:134]: Yes, but Getting all changed delete files from iceberg's entries table is also the same for Arctic and Iceberg Table as I  know, and they share many deplicate code now.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**